tests/catalog/test_rest.py

```diff
@@ -560,6 +560,64 @@ def test_create_table_409(rest_mock: Mocker, table_schema_simple: Schema) -> Non
     assert "Table already exists" in str(e.value)
 
 
+def test_create_table_if_not_exists_200(
```

Comment:
Nice! It would have been easier to write an integration test, but great job on the mocking 👍